pkgs/development/python-modules/samsungctl/default.nix

```diff
@@ -11,12 +15,17 @@ buildPythonPackage rec {
     sha256 = "0ipz3fd65rqkxlb02sql0awc3vnslrwb2pfrsnpfnf8bfgxpbh9g";
   };
 
-  propagatedBuildInputs = [
-    websocket-client
-  ];
+  passthru.extras-require = {
+    websocket = [
+      websocket-client
+    ];
+    # interactive_ui requires curses package
+  };
 
   # no tests
-  doCheck = false;
+  doCheck = true;
```

Comment:
`checkPhase` should be specified or `pytestCheckHook` added to `checkInputs`
no need to specify `doCheck = true`

Reply:
Yeah, just leftovers from when I was checking out the tests.

Reply:
Shouldn't the dependencies be in propagatedBuildInputs if they're required for importing the module?

Reply:
Spot on.